### DISCUSSION TITLE
Importer doesn't seem to read config

### DISCUSSION THREAD

**Discussion by tjkolev** [firefly-iii/firefly-iii#9287] [General]
Greetings!

Installed Firefly-III + Importer via Docker. Only changes made to the docker-compose.yml are the port bindings:
```
services:
  app:
    ports:
      - '8800:8080'

  importer:
    ports:
      - '8801:8080'
```

The docker images are running on my server, and I use the application from my laptop. Firefly works fine at http://komodo:8800. And http://komodo:8801/token also comes up. But then the Submit fails with ERR_CONNECTION_REFUSED. Because it's trying to go to http://localhost:8080/oauth/authorize. That won't work on my laptop.

Things of note.
(1) On http://komodo:8801/token it says this:
```
In order to get access to your Firefly III installation at http://app:8080/ (http://localhost:8080/) , you will need to submit a valid Client ID.
```
The second URL isn't correct.

(2) This is what I have in `.importer.env`:
```
FIREFLY_III_URL=http://app:8080
VANITY_URL=http://komodo:8800
```

(3) This is the importer log upon hitting "Submit":
```
[2024-09-30 03:10:21] local.DEBUG: [a] Base URL is "http://app:8080"
[2024-09-30 03:10:21] local.DEBUG: [b] Vanity URL is now "http://app:8080"
[2024-09-30 03:10:21] local.DEBUG: [c] Vanity URL is now "http://localhost:8080"
[2024-09-30 03:10:21] local.DEBUG: Now in App\Http\Controllers\TokenController::redirectForPermission(request, "http://app:8080", "http://localhost:8080", 2)
[2024-09-30 03:10:21] local.DEBUG: Now redirecting to "http://localhost:8080/oauth/authorize?" (params omitted)
```

No matter what I set for VANITY_URL, the importer is always using http://localhost:8080. I stop/start the container when updating .importer.env. I expect it would redirect to http://komodo:8800.

I've read https://docs.firefly-iii.org/how-to/data-importer/how-to-configure/ and the table for FIREFLY_III_URL and VANITY_URL, but that (and all) documentation cover the case when using the application on the installation machine. I'd say more common would be using it on a different machine. And I couldn't figure it out.

What am I missing to have the importer authenticate and run properly? I'd appreciate any suggestions.

Cheers!
tjk :)

**Comment by JC5** [comment]
Does it work after you press "start over"?

**Reply by tjkolev** [reply]
Nope, it does not. "start over" changes nothing.
tjk :)

**Reply by JC5** [reply]
Sorry, that's all the ideas I have. The container doesn't cache anything itself, but old information can be retained in cookies. Otherwise, you can try deleting the container rather than start/stopping it. I'm not sure when/if Docker resets environment variables.

**Reply by tjkolev** [reply]
Removing the container, and re-creating it solved the problem. So just restarting it, doesn't make it re-read the .importer.env file. Funky...

Cheers!
tjk :)
